### PR TITLE
feat(playground): add playground/preview of your DocSearch index

### PR DESCRIPTION
**Pull request by s-pace**
This PR add a new feature: the playground that enables anyone to try out his DocSearch implementation:
```
https://docsearch.algolia.com/playground/?indexName=<indexName>&apiKey=<apiKey>
```

Optional parameter: `appId`

### CODE REVIEW

**Comment by s-pace** [comment]
This PR add a new feature: the playground that enables anyone to try out his DocSearch implementation:
```
https://docsearch.algolia.com/playground/?indexName=<indexName>&apiKey=<apiKey>
```

Optional parameter: `appId`

**Comment by s-pace** [comment]
Follow #7

**Comment by francoischalifour** [comment]
Before reviewing the code, there are a few things that need to be addressed:

1. When the credentials are wrong, it displays the following error message without the search documentation.

> The credentials provided from the URL were wrong, we will demo the search with the search of our documentation instead.

2. Why do the `h2` and `p` display the same content?

3. We should display the dropdown at the top of the page because it's cropped.

<details>
<summary>See screenshot</summary>

image

</details>

4. Should we add a "Playground" link in the navbar?

**Comment by s-pace** [comment]
Thanks @francoischalifour 

>1

It does work. Here is an example on the local version:
image

> 2
✅ 

>3
Good point, ✅ 
>4
Sure, ✅

File: src/components/DocSearch/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        indexName,
+        inputSelector: `#${docsearchRef.current.id}`,
+        debug: true, // Set debug to true if you want to inspect the dropdown
+      });
+    });
+  }, [docsearchRef, indexName, apiKey]);
+
+  return (
+    <>
+      <link
```
Comment (francoischalifour) [inline src/components/DocSearch/index.js:26]: This should go in a [`Head`](https://v2.docusaurus.io/docs/docusaurus-core/#head).

File: src/components/ErrorBoundary.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    super(props);
+    this.state = {
+      hasError: false,
+      error: null,
+      info: null,
+    };
+  }
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    console.error('getDerivedStateFromError', error);
```
Comment (francoischalifour) [inline src/components/ErrorBoundary.js]: We should remove the `console` occurrences.

File: src/components/ErrorBoundary.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    return { hasError: true, error };
+  }
+  componentDidCatch(error, info) {
+    console.error('componentDidCatch', error);
```
Comment (francoischalifour) [inline src/components/ErrorBoundary.js]: We should remove the `console` occurrences.

File: src/pages/playground.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import Layout from '@theme/Layout';
+import { useLocation } from 'react-router';
+import queryString from 'query-string';
+import { LabelText, Hero, Text } from '@algolia/ui-library';
+import DocSearch from '../components/DocSearch';
+import ErrorBoundary from '../components/ErrorBoundary';
+import algoliasearch from 'algoliasearch';
```
Comment (francoischalifour) [inline src/pages/playground.js:8]: We should load only the client with search capabilities, this one also handles indexing and is heavier.

```suggestion
import algoliasearch from 'algoliasearch/lite';
```

File: src/pages/playground.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    // Credential not provided
+    if (!indexName && !apiKey) {
+      fallbackToDocSearchDocCred();
+      return;
+    }
+    if ((!indexName && !apiKey) || apiKey.length !== 32) {
+      setWrongCredentials(true);
+      fallbackToDocSearchDocCred();
+      return;
+    }
```
Comment (francoischalifour) [inline src/pages/playground.js:41]: Do we need to handle this ourselves or can be rely on the `index.search('')` check?

File: src/components/DocSearch/index.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    if (!docsearchRef.current) {
+      return;
+    }
+    import('docsearch.js').then(({ default: docsearch }) => {
+      docsearch({
+        appId,
+        apiKey,
+        indexName,
+        inputSelector: `#${docsearchRef.current.id}`,
+        debug: true, // Set debug to true if you want to inspect the dropdown
```
Comment (francoischalifour) [inline src/components/DocSearch/index.js]: Since the input is not styled, what about leveraging `enhancedSearchInput`?

**Review by francoischalifour — COMMENTED** [review]
The highlighted parts are not vertically aligned with the non-highlighted parts:

image

I think there's an unwanted gradient:

image

The experience is buggy when credentials are wrong. I think we don't need to fallback, but rather to display the link that goes to the playground with the default DocSearch playground.

ds-playground-bug

File: src/components/DocSearch/index.js (hunk 6 above)
Comment (Shipow) [reply src/components/DocSearch/index.js]: enhancedSearchInput is broken in the latest versions, we'll do some skinning with ui-lib

**Comment by Shipow** [comment]
Made some skinning
image
But I'd like to see the instructions (snippets + faq) just bellow the try out card.

**Review by Shipow — APPROVED** [review]
Good to go, but need to be completed with the install instructions bellow the demo (dedicated snippet, faq, etc..).

File: src/pages/playground.js (hunk 5 above)
Comment (s-pace) [reply src/pages/playground.js:41]: Faster check, let's keep it.

File: src/pages/playground.js (hunk 4 above)
Comment (s-pace) [reply src/pages/playground.js:8]: will do

**Pull request merged**